File: tests/machine/x/dart/sort_stable.dart
import 'dart:convert';

var items = [
  {'n': 1, 'v': 'a'},
  {'n': 1, 'v': 'b'},
  {'n': 2, 'v': 'c'},
];

var result = (() {
  var _q0 = <dynamic>[];
  for (var i in items) {
    _q0.add([i['n'], i['v']]);
  }
  _q0.sort(
    (a, b) => (jsonEncode(a[0]) as Comparable).compareTo(jsonEncode(b[0])),
  );
  _q0 = [for (var x in _q0) x[1]];
  return _q0;
})();

void main() {
  print(result);
}

File: tests/machine/x/cs/sort_stable.cs
using System;
using System.Collections.Generic;
using System.Linq;

class Program {
    static void Main() {
        var items = new List<dynamic> { new Dictionary<dynamic, dynamic> { { "n", 1 }, { "v", "a" } }, new Dictionary<dynamic, dynamic> { { "n", 1 }, { "v", "b" } }, new Dictionary<dynamic, dynamic> { { "n", 2 }, { "v", "c" } } };
        List<string> result = items.OrderBy(i => i.n).Select(i => i.v).ToArray();
        Console.WriteLine("[" + string.Join(", ", result) + "]");
    }
}
